opsdroid/connector/gitter/__init__.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import aiohttp
+import asyncio
+import json
+import urllib
+
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message
+
+_LOGGER = logging.getLogger(__name__)
+GITTER_STREAM_API = "https://stream.gitter.im/v1/rooms"
+GITTER_MESSAGE_BASE_API = "https://api.gitter.im/v1/rooms"
+
+
+class ConnectorGitter(Connector):
+    """A connector for Gitter."""
+
+    def __init__(self, config, opsdroid=None):
+        """Create the connector."""
+        super().__init__(config, opsdroid=opsdroid)
+        _LOGGER.debug("Starting Gitter connector")
+        self.name = "gitter"
+        self.session = None
+        self.response = None
+        self.bot_name = self.config.get("bot-name", "opsdroid")
+        self.room_id = self.config.get("room-id")
+        self.access_token = self.config.get("access-token")
+        self.update_interval = 1
+        self.opsdroid = opsdroid
+        self.listening = True
+
+    async def connect(self):
+        """Create the connection."""
+
+        # Create connection object with chat library
+        _LOGGER.debug("Connecting with gitter stream")
+        self.session = aiohttp.ClientSession()
+        gitter_url = self.build_url(
+            GITTER_STREAM_API,
+            self.room_id,
+            "chatMessages",
+            access_token=self.access_token,
+        )
+        self.response = await self.session.get(gitter_url, timeout=None)
+
+    def build_url(self, base_url, *res, **params):
+        """Build the url. args ex:(base_url,p1,p2=1,p2=2)."""
+
+        url = base_url
+        for r in res:
+            url = "{}/{}".format(url, r)
+        if params:
+            url = "{}?{}".format(url, urllib.parse.urlencode(params))
+        return url
+
+    async def listen(self):
+        """Keep listing to the gitter channel."""
+        _LOGGER.debug("Listening with gitter stream")
+        while self.listening:
+            try:
+                await self._get_messages()
+            except AttributeError:
+                break
+
+    async def _get_messages(self):
+        """Message listener."""
+        await asyncio.sleep(self.update_interval)
+        async for data in self.response.content.iter_chunked(1024):
+            message = await self.parse_message(data)
+            if message is not None:
+                await self.opsdroid.parse(message)
+
+    async def parse_message(self, message):
+        """Parse response from gitter to send message."""
+        message = message.decode("utf-8").rstrip("\r\n")
+        if len(message) > 1:
+            message = json.loads(message)
+            _LOGGER.debug(message)
+            try:
+                return Message(
+                    message["text"], message["fromUser"]["username"], self.room_id, self
+                )
+            except KeyError as err:
+                _LOGGER.error("Unable to parse message %s", message)
+                _LOGGER.debug(err.with_traceback())
```

Comment:
The method `with_traceback` is used to set the traceback, not get it. I would just log the whole error object.

```suggestion
                _LOGGER.debug(err)
```

Reply:
As per my understanding it print all stack trace.But yes we can print err object as well.

Reply:
[I don't think it does.](https://docs.python.org/3/library/exceptions.html#BaseException.with_traceback) I think you are thinking of `print_exception` from the [traceback library](https://docs.python.org/3/library/traceback.html#traceback.print_exception).